src/solver/hydro/daily/h2o_j_lisser_les_sur_turbines.cpp

```diff
@@ -32,94 +32,83 @@
 
 void H2O_J_LisserLesSurTurbines(DONNEES_MENSUELLES* DonneesMensuelles, int NumeroDeProbleme)
 {
-    int Pdt;
-    int Cnt;
-    int NbPdt;
-    double X;
-    int Np;
-    char* Flag;
-    char LimiteAtteinte;
-    int NbCycles;
-    double SurTurbineARepartir;
-    double Xmoy;
-    double MargeMin;
-    double SurTurbine;
     PROBLEME_HYDRAULIQUE& ProblemeHydraulique = DonneesMensuelles->ProblemeHydraulique;
 
-    std::vector<double>& TurbineMax = DonneesMensuelles->TurbineMax;
-    std::vector<double>& Turbine = DonneesMensuelles->Turbine;
-    std::vector<double>& TurbineCible = DonneesMensuelles->TurbineCible;
+    auto& Turbine = DonneesMensuelles->Turbine;
+    const auto& TurbineMax = DonneesMensuelles->TurbineMax;
+    const auto& TurbineCible = DonneesMensuelles->TurbineCible;
 
-    Flag = (char*) ProblemeHydraulique.ProblemeLineairePartieVariable[NumeroDeProbleme].Xmax.data();
+    const int NbPdt = ProblemeHydraulique.NbJoursDUnProbleme[NumeroDeProbleme];
 
-    NbPdt = ProblemeHydraulique.NbJoursDUnProbleme[NumeroDeProbleme];
+    std::vector<bool> flag(NbPdt);
 
-    SurTurbineARepartir = 0.0;
-    for (Pdt = 0; Pdt < NbPdt; Pdt++)
+    double SurTurbineARepartir = 0.0;
+    for (int Pdt = 0; Pdt < NbPdt; Pdt++)
     {
-        Flag[Pdt] = 0;
+        flag[Pdt] = 0;
         if (Turbine[Pdt] - TurbineCible[Pdt] > ZERO)
-            Flag[Pdt] = 1;
+            flag[Pdt] = 1;
     }
 
-    for (Pdt = 0; Pdt < NbPdt; Pdt++)
+    for (int Pdt = 0; Pdt < NbPdt; Pdt++)
     {
-        if (Flag[Pdt] == 1)
+        if (flag[Pdt] == 1)
             SurTurbineARepartir += Turbine[Pdt] - TurbineCible[Pdt];
     }
 
-    for (Pdt = 0; Pdt < NbPdt; Pdt++)
-        Flag[Pdt] = 0;
-    for (Pdt = 0; Pdt < NbPdt; Pdt++)
+    for (int Pdt = 0; Pdt < NbPdt; Pdt++)
+        flag[Pdt] = 0;
+    for (int Pdt = 0; Pdt < NbPdt; Pdt++)
     {
         if (TurbineMax[Pdt] - TurbineCible[Pdt] > ZERO)
-            Flag[Pdt] = 1;
+            flag[Pdt] = 1;
     }
 
-    NbCycles = 0;
+    int NbCycles = 0;
 BoucleDeRepartition:
 
-    Np = 0;
-    for (Pdt = 0; Pdt < NbPdt; Pdt++)
+    int Np = 0;
+    for (int Pdt = 0; Pdt < NbPdt; Pdt++)
```

Comment:
```suggestion
    int Np = std::count(flag.begin(), flag.end(), true);
```

Reply:
You could even use `std::none_of`, since Np is only used to determine if one of `flag[i]` is `true`. We don't seem to care how many are true.